Quest pitches with unknown sport

PanierAvide opened the pull request:
See #180

westnordost commented on app/src/main/java/de/westnordost/streetcomplete/quests/sport/AddSportForm.java:
```diff
@@ -0,0 +1,150 @@
+			this.osmValue = osmValue;
+		}
+	}
+}
```
There is quite some copy&paste code here from roof shape form. The first oppurtunity is always the best opportunity to refactor something.
Are you confident you can do that? Otherwise I would do this myself after the merge.

PanierAvide replied:
I will try to do the refactor.

westnordost commented:
I noticed that for `sport=...`, there are many tags that look like this: I.e `sport=australian_football;cricket`
While looking it up, I found out that the ";" to specify several sports is explicitly allowed (and is thousand times more meaningful than `multi`).

As I understand [the wiki](https://wiki.openstreetmap.org/wiki/Tag%3Asport%3Dmulti ), multi is intended primarily for pitches that are "not intended for any specific sport", so basically "pitches" that are basically just a big rectangular area with grass on it intended for playing games.

I suggest to change _"Several sports can be played"_ into _"Not any specific sport"_, _"Which sport can be played here?"_ into _"Which sport **is** played here?"_ and make it possible to select multiple sports (add a hint that you can select several). As a safeguard against misunderstandings (or the too keen), there could be a "are you sure" dialog when the user selects more than 3 different sports, hinting him to the possibility to also answer with `multi `if it is not intended for any specific sport.

PanierAvide commented on app/src/main/assets/streetcomplete.yaml:
```diff
@@ -30,7 +30,7 @@ textures:
             max_speed: [576,576,192,192]
             max_weight: [768,576,192,192]
-            #empty: [960,576,192,192]
+            sport: [960,576,192,192]
```
Yes of course you can make it better 😉

PanierAvide commented on app/src/main/java/de/westnordost/streetcomplete/quests/sport/AddSport.java:
```diff
@@ -0,0 +1,48 @@
+	@Override
+	protected String getTagFilters()
+	{
+		return "nodes, ways with leisure=pitch and !sport";
```
I did not look for relations as they take much longer time to retrieve against Overpass API and they represent less than 0.5% of all pitches. So pitches relations without sport tag are only a few (for example 44 over all France), I'm not sure it is worth the look-up cost.

Why not check for deprecated tags, but this shouldn't be another quest ?

westnordost replied:
> multipolygons

Ah ok, I see

> Why not check for deprecated tags, but this shouldn't be another quest ?

Because, how would that quest look like to the user? It doesn't make sense to inform the user that this pitch has been tagged with a deprecated tag and so he should resurvey it etc pp. To the user, the reasons why this quest is displayed are irrelevant - he should just answer the question.

So, it would look exactly the same as this one and also behave the same: . The idea is that the quest would just treat pitches that are tagged as team_handball or hockey as if it hasn't been tagged yet. This is not so important for team_handball, as this is really just a synonym but more so with hockey, because in the Americas, with "hockey" probably ice hockey is meant while elsewhere, probably field hockey is meant. These two can be distinguished at first glance ( ice / no ice).

PanierAvide replied:
OK, it makes sense to do it that way.

westnordost commented:
I will implement a MultiImageSelectAdapter today which can be used to check several things.

PanierAvide commented:
I have added the sport names in the list, changed labels, and merged some code between roof shape and sport quests into ImageListQuestAnswerFragment. I let you work on MultiImageSelectAdapter as you have already started, maybe single/multi image selector can be set as an option when using ImageListQuestAnswerFragment.

westnordost commented:
Nice! There are a few small things I'd change, but I think I will simply do them myself after the merge, this is quicker then discussing it.

I will also merge right now, because the changes I did for the ImageMultiSelectadapter will lead to conflicts that I don't want to impose upon you.

PanierAvide commented:
OK great, thanks for merging.

westnordost commented:
![device-2017-05-14-213700](https://cloud.githubusercontent.com/assets/4661658/26037141/9f8bc3ca-38ed-11e7-83fc-030b79d50f6c.png)
Isn't it a beauty? :-)
(changed the cell layout and wording a bit, commit in a few minutes)

westnordost commented:
https://github.com/westnordost/StreetComplete/commit/c59c774e179d51d9210cc37f4d35b0b0cdb9dc4e
https://github.com/westnordost/StreetComplete/commit/aee0cabeec77c520d01098ddca17c96cbf07c96e
https://github.com/westnordost/StreetComplete/commit/1153d11f909117611bc3c5f0677e64ca1e4908d1

westnordost commented:
And added the terms to [POEditor](https://poeditor.com/projects/view?id=97843)

PanierAvide commented:
It looks great ! I have added the French translations in POEditor. Thanks for the refactoring work and multi image selector.
